.travis.yml

```diff
@@ -47,26 +49,97 @@ _base_envs:
       eval "$(pyenv init -)"
       eval "$(pyenv virtualenv-init -)"
     fi
-  - pyenv install --skip-existing --keep --verbose "$PYPY_VERSION"
-  - pyenv shell "$PYPY_VERSION"
-  - python --version
+  - &install_python pyenv install --skip-existing --keep --verbose "$PYTHON_VERSION"
+  - &switch_python pyenv shell "$PYTHON_VERSION"
+  - &python_version python --version
+- &linux_python_base
+  <<: *linux_base
+  <<: *pyenv_base
+- &osx_python_base
+  <<: *pyenv_base
+  os: osx
+  language: generic
+  before_install:
+  - brew update
+  - brew install readline xz
+  - *ensure_pyenv_installed
+  - *install_python
+  - *switch_python
+  - *python_version
+  before_cache:
+  - brew --cache
 matrix:
   fast_finish: true
   allow_failures:
   - env: TOXENV=pre-commit-pep257
   include:
-  - <<: *pypy_base
+  - <<: *linux_python_base
     python: pypy
     env:
-    - PYPY_VERSION=pypy2-5.7.1
-    - PYENV_ROOT="$HOME/.pyenv"
-    - PATH="$PYENV_ROOT/bin:$PATH"
-  - <<: *pypy_base
+    - PYTHON_VERSION=pypy2-5.7.1
+    - *env_pyenv
+    - *env_path
+  - <<: *linux_python_base
     python: pypy3
     env:
-    - PYPY_VERSION=pypy3.5-5.7.1-beta
-    - PYENV_ROOT="$HOME/.pyenv"
-    - PATH="$PYENV_ROOT/bin:$PATH"
+    - PYTHON_VERSION=pypy3.5-5.7.1-beta
+    - *env_pyenv
+    - *env_path
+  - <<: *osx_python_base
+    python: 2.6
+    env:
+    - PYTHON_VERSION=2.6.9
+    - *env_pyenv
+    - *env_path
+  - <<: *osx_python_base
+    python: 2.7
+    env:
+    - PYTHON_VERSION=2.7.13
+    - *env_pyenv
+    - *env_path
+  - <<: *osx_python_base
+    python: 3.3
+    env:
+    - PYTHON_VERSION=3.3.6
+    - *env_pyenv
+    - *env_path
+  - <<: *osx_python_base
+    python: 3.4
+    env:
+    - PYTHON_VERSION=3.4.6
+    - *env_pyenv
+    - *env_path
+  - <<: *osx_python_base
+    python: 3.5
+    env:
+    - PYTHON_VERSION=3.5.3
+    - *env_pyenv
+    - *env_path
+  - <<: *osx_python_base
+    python: 3.6
+    env:
+    - PYTHON_VERSION=3.6.1
+    - *env_pyenv
+    - *env_path
+  - <<: *osx_python_base
+    python: nightly
+    env:
+    - PYTHON_VERSION=3.7-dev
+    - *env_pyenv
+    - *env_path
+  - <<: *osx_python_base
+    python: pypy2-5.7.1
+    env:
+    - PYTHON_VERSION=pypy2-5.7.1
+    - *env_pyenv
+    - *env_path
+  # Fails under OS X:
```

Comment:
I generally try to avoid commented code. I suggest either marking this test as an allowed failure (if you intend for it to work in the near future) or just remove it (and let the source code history remember it in case we choose to revive it).